[#3449] Document co-sign flow (#3567)
open-formulieren/open-forms: docs/manual/forms/cosign_flow.rst

.. _manual_cosign_flow:

> Mede-ondertekenen
> =================
> 
> Flow
> ----

Reviewer: ```suggestion
Mede-ondertekenen
=================

Open Formulieren ondersteunt formulieren waarin een tweede persoon een inzending moet ondertekenen voor deze
verwerkt wordt, zoals een partner of voogd. Er is dus een extra persoon betrokken naast de invuller van het formulier.

Flow
----
```

I'd give a little introduction about what cosigning is and why the need could exist for that.

De mede-ondertekenen flow is hieronder beschreven. Persoon 1 is de persoon die als eerste het formulier invult.
Persoon 2 is de mede-ondertekenaar.

#. Persoon 1 logt in om het formulier te beginnen.

> #. In het formulier bevindt zich een "Mede-ondertekenen" component. Hier kan persoon 1 de email adres invoeren
>    van persoon 2.

Reviewer: ```suggestion
#. In het formulier bevindt zich een "Mede-ondertekenen" component. Hier kan persoon 1 het e-mailadres van persoon 2 opgeven.
```

> #. Op het moment dat het formulier wordt ingestuurd, krijgt persoon 1 een bevestigingsemail (als dit
>    geconfigureerd is in het formulier).

Reviewer: ```suggestion
#. Zodra het formulier ingestuurd wordt, krijgt persoon 1 een bevestigingsmail (als deze ingeschakeld is).
```
"bevestigingsmail" or "bevestigings-e-mail" are both correct spellings, I prefer the former because the latter looks weird.

> #. Persoon 2 krijgt een email met het verzoek om het formulier te mede-ondertekenen. Het is mogelijk om in de
>    **Algemene configuratie** te configureren of deze email de link naar het formulier bevat of niet.

Reviewer: ```suggestion
#. Persoon 2 ontvangt een e-mail met het verzoek om het formulier te mede-ondertekenen. De e-mail 
    bevat of een directe link óf instructies om het formulier te openen (afhankelijk van de instelling in de 
    **Algemene configuratie**). In de e-mail staat een verificatiecode voor de inzending.
```

> #. Persoon 2 moet naar de pagina van het formulier navigeren en inloggen door op de knoppen onder
>    **Log in om het formulier mede te ondertekenen** te clicken.

Reviewer: ```suggestion
#. Persoon 2 navigeert naar de startpagina van het formulier navigeren en logt in met de 
**Log in om het formulier mede te ondertekenen**-knop.
```

   .. image:: _assets/cosign_buttons.png
       :width: 100%

> #. Dan moet persoon 2 de inzending code invullen die per email was ontvangen. Als dit gedaan is, worden de gegevens
>    van de inzending getoond.

Reviewer: ```suggestion
#. Vervolgens moet persoon 2 de verificatiecode uit de e-mail invullen. Bij een geldige code ziet de gebruiker een overzicht van de ingevulde gegevens.
```

> #. Als de gegevens correct zijn, persoon 2 moet eerst het privacy beleid accepteren en/of de verklaring van waarheid en
>    daarna medeondertekenen.

Reviewer: ```suggestion
#. Indien relevant, dan moet persoon 2 het privacybeleid en/of de verklaring van waarheid accepteren (dit is een formulierinstelling). Daarna bevestigt de gebruiker de mede-ondertekening.
```

> #. Beide persoon 1 en persoon 2 krijgen dan een bevestigings email.

Reviewer: ```suggestion
#. Persoon 1 en persoon 2 ontvangen daarna allebei een bevestigingsmail.
```

> #. De inzending wordt verwerkt (naar de registratie backend gestuurd) pas nadat die medeondertekend is.

Reviewer: ```suggestion
#. De inzending wordt nu verwerkt met de gekoppelde registratieplugin (dus pas ná het mede-ondertekenen).
```
